backend/mobilityAI/Controllers/MobilityAIController.cs

```diff
@@ -308,19 +400,99 @@ public IActionResult MlCallback(string Id, IFormFile Activities)
         }
 
         /// <summary>
-        /// Gets processed activity data for a specific device in a specific time range
+        /// Gets processed activity data for a specific device in a specific time range to be displayed on the UI
         /// </summary>
         /// <param name="Start">Epoch for beginning of time range</param>
         /// <param name="End"Epoch for end of time rante></param>
         /// <param name="patientId">Patient Id for the data you want to query</param>
         /// <returns></returns>
         [HttpGet("GetActivityData", Name = "GetActivityData")]
-        public IActionResult GetActivityData(long Start, long End, int patientId)
+        public IActionResult GetActivityData(long start, long end, int patientId)
         {
             var data = (from activities in _context.Activities
-                        where activities.Start >= Start && activities.End <= End && activities.PatientId == patientId
-                        select new { activities.Start, activities.End, activities.Type }).ToList();
-            return Ok(JsonConvert.SerializeObject(data));
+                        where activities.Start >= start && activities.End <= end && activities.PatientId == patientId
+                        orderby activities.Start ascending
+                        select new { activities.Start, activities.End, activities.Type })
+                        .ToList();
+
+            if(data.Count != 0) {
+
+                float[] count = new float[5];
+                float[] total = new float[5];
+                float[][] activityTotals = new float[5][];
+                for (int i = 0; i < 5; i++) activityTotals[i] = new float[13];
+                float totalRows = data.Count;                
+
+                // long firstHour = data[0].Start;
+                // long lastHour = data[data.Count-1].End;
+        
+                // long diff = lastHour - firstHour;
+
+                for(int i = 0; i < data.Count-1; i++) {
+                    var element = data[i];
+                    var nextElement = data[i+1];
+
+                    count[element.Type]++;
+
+                    int startHour = (FromUnixTime(element.Start/1000).Hour) - 12;
```

Comment:
Store `element.Start` in a `DateTime` like they do [here](https://stackoverflow.com/a/963812/5086919) then get the local hour with `.ToLocalTime()`

Reply:
Getting the local time would have the same effect as doing this just with a different offset.

Reply:
Also requires changes to react native parts, will update once we rework the API